packages/nextjs-mf/src/plugins/container/InvertedContainerRuntimeModule.ts

```diff
@@ -1,824 +1,110 @@
-        ],
-      )}`,
-    ]);
+    const entryModules =
+      this.compilation.chunkGraph.getChunkEntryModulesIterable(container);
+    return Array.from(entryModules)[0];
   }
 
-  /**
-   * Generate method for the runtime module, producing the runtime code.
-   * @returns {string} runtime code
-   */
-  override generate() {
+  override generate(): string {
     if (!this.compilation || !this.chunk || !this.chunkGraph) {
       return '';
     }
 
-    const { compilation } = this;
-
-    const { name } = this.options;
-    const { chunkGraph, chunk } = this;
-    const { compiler } = compilation;
-
-    const { RuntimeGlobals, Template, javascript } = compiler.webpack || {};
-    const chunkHasJs =
-      (javascript && javascript.JavascriptModulesPlugin.chunkHasJs) ||
-      require('webpack/lib/javascript/JavascriptModulesPlugin').chunkHasJs;
-
+    const { name, debug } = this.options;
     const containerEntryModule = this.resolveContainerModule() as
       | (Module & { _name: string })
       | undefined;
-    //server runtime is always called webpack-runtime
-    const isServer = chunk.name === 'webpack-runtime';
-    const isApi = chunk.name === 'webpack-api-runtime';
-    const conditionMap = chunkGraph.getChunkConditionMap(chunk, chunkHasJs);
-    // const hasJsMatcher = compileBooleanMatcher(conditionMap);
-    // find the main webpack runtime, skip all other chunks
-    if (chunk.name !== 'webpack-runtime' && chunk.name !== 'webpack') {
-      return Template.asString('');
+    const containerName = containerEntryModule?._name || name;
+    const chunk = this.chunk;
+
+    const containerModuleId =
+      containerEntryModule?.id || containerEntryModule?.debugId;
+    // const hasEnsurechunkHandlers = !this.compilation.runtimeRequirements.has(RuntimeGlobals.ensureChunkHandlers)) {
+    const isPartialContainer = chunk.runtime === this.options.runtime;
+    if (!(containerName && containerModuleId) || !isPartialContainer) {
+      return '';
     }
-
-    const containerEntry = [containerEntryModule].map((module) => {
-      const containerName = module?._name || name;
-      const containerModuleId = module?.id || module?.debugId;
-
-      if (!(containerName && containerName)) {
-        return '';
+    const globalObject = `globalThis.__remote_scope__`;
+    const containerScope = `${RuntimeGlobals.global}`;
+
+    return `
+      function attachRemote (resolve) {
+        const innerRemote = __webpack_require__(${JSON.stringify(
+          containerModuleId,
+        )});
+        console.log({innerRemote, name: ${RuntimeGlobals.runtimeId}})
+        if(${globalObject} && !${globalObject}[${JSON.stringify(name)}]) {
+          ${globalObject}[${JSON.stringify(name)}] = innerRemote;
+        } else if(${containerScope} && !${containerScope}[${JSON.stringify(
+          name,
+        )}]) {
+          ${containerScope}[${JSON.stringify(name)}] = innerRemote;
+        }
+        __webpack_require__.S.default = new Proxy({}, {
```

Comment:
@2heal1 this is workaround but allows share scope to prefer host modules and control shared.